File: iosApp/iosApp/views/navigation/bars/Level1NavigationRail.swift
//
//  Level1NavigationRail.swift
//  iosApp
//
//  Created by Daniele Baroncelli on 24/05/21.
//  Copyright © 2021 orgName. All rights reserved.
//

import SwiftUI
import shared


extension Navigation {

    // this is the left vertical navigation bar for 2-Pane visualization
    // (used by bigger devices in landscape mode)
    
    @ViewBuilder func level1NavigationRail(selectedTab: ScreenIdentifier) -> some View {

        Spacer()
        NavigationRailButton(
            itemLabel: "All Countries",
            iconName: "list.bullet",
            selected: selectedTab.URI==Level1NavigationImpl.allcountries.screenIdentifier.URI,
            onClick: { self.navigateByLevel1Menu(level1NavigationItem: Level1NavigationImpl.allcountries) }
        )
        NavigationRailButton(
            itemLabel: "Favorites",
            iconName: "star.fill",
            selected: selectedTab.URI==Level1NavigationImpl.favoritecountries.screenIdentifier.URI,
            onClick: { self.navigateByLevel1Menu(level1NavigationItem: Level1NavigationImpl.favoritecountries) }
        )
        Spacer()

    }
        
}




struct NavigationRailButton: View {
    var itemLabel : String
    var iconName : String
    var selected : Bool
    var onClick : () -> Void
    
    var body: some View {
        Button(action: { onClick() }) {
            VStack(spacing: 5) {
                Image(systemName: iconName).resizable().scaledToFit().frame(height:15)
                Text(itemLabel).font(Font.footnote)
            }
            .padding(.top, 25)
            .padding(.bottom, 25)
            .foregroundColor(selected ? .white : linkColor)
        }
    }
}

File: androidApp/src/main/java/com/fieldontrack/kmm/android/composables/navigation/bars/Level1NavigationRail.kt
package com.fieldontrack.kmm.android.composables.navigation.bars

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.Icon
import androidx.compose.material.LocalContentColor
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Menu
import androidx.compose.material.icons.filled.Star
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.fieldontrack.kmm.feature.core.Navigation
import com.fieldontrack.kmm.feature.core.ScreenIdentifier
import com.fieldontrack.kmm.navigation.Level1NavigationImpl


// this is the left vertical navigation bar for 2-Pane visualization
// (used by bigger devices in landscape mode)

@Composable
fun Navigation.Level1NavigationRail(
    selectedTab: ScreenIdentifier
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .fillMaxHeight()
            .background(MaterialTheme.colors.primary),
        verticalArrangement = Arrangement.Center
    ) {
        NavigationRailItem(
            icon = { Icon(Icons.Default.Menu, "ALL") },
            label = { Text("All Countries", fontSize = 13.sp) },
            selected = selectedTab.URI == Level1NavigationImpl.AllCountries.screenIdentifier.URI,
            onClick = { navigateByLevel1Menu(Level1NavigationImpl.AllCountries) }
        )
        NavigationRailItem(
            icon = { Icon(Icons.Default.Star, "FAVORITES") },
            label = { Text("Favourites", fontSize = 13.sp) },
            selected = selectedTab.URI == Level1NavigationImpl.FavoriteCountries.screenIdentifier.URI,
            onClick = { navigateByLevel1Menu(Level1NavigationImpl.FavoriteCountries) }
        )
    }
}


@Composable
fun ColumnScope.NavigationRailItem(
    icon: @Composable () -> Unit,
    label: @Composable () -> Unit,
    selected: Boolean,
    onClick: () -> Unit
) {
    CompositionLocalProvider(
        LocalContentColor provides if (selected) MaterialTheme.colors.background else MaterialTheme.colors.primaryVariant
    ) {
        Row(
            modifier = Modifier
                .clickable { onClick() }
                .padding(top = 25.dp, bottom = 25.dp)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth(),
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                icon()
                label()
            }
        }
    }
}
